extension.js

```diff
@@ -1,57 +1,160 @@
+    }
+
     /**
      * Get a require or define statement from string
      * @param {String} str
      * @return {String} 
      */
     getRequireOrDefineStatement(str) {
-        const match = /(define|require)\s*\(([^)]*)/gi.exec(str + "");
+        str = this.stripWhiteSpaceAndNewLines(str);
+        const match = /(define|require).*{/i.exec(str);
         return match && match[0] || null;
     }
 
     /**
+     * Get partial code from require/define statement until given line and character
+     * @param {VSCODE Document} document
+     * @param {Int} line to stop
+     * @param {Int} characterIndex to stop
+     * @return {String} substring for given document ending at given line and character and starting at last define or require
+     */
+    getRequireOrDefineCodeUntilCharacterIndex(document, line, characterIndex) {
+        const textBeforeChar = document.getText(new Range(0, 0, line, characterIndex));
+        const textBeforeCharWithoutComments = this.stripAllComments(textBeforeChar);
+        const lastOccuranceDefine = textBeforeCharWithoutComments.toLowerCase().lastIndexOf('define');
+        const lastOccuranceRequire = textBeforeCharWithoutComments.toLowerCase().lastIndexOf('require');
+        const lastOccuranceRequireOrDefine = lastOccuranceDefine > lastOccuranceRequire ? lastOccuranceDefine : lastOccuranceRequire;
+
+        return textBeforeCharWithoutComments.substr(lastOccuranceRequireOrDefine > -1 ? lastOccuranceRequireOrDefine : 0);
+    }
+
+    /**
+     * Check if string contains multiple define or require statements
+     * @param {String} str to search in 
+     * @return {Bool}
+     */
+    stringHasMultipleDefineOrRequireStatements(str) {
+        return (str.match(/^\s*(define|require)\(/gm) || []).length > 1;
+    }
+
+    /**
+     * Check if needle is a module (path) in define or require statement.
+     * @param {String} needle to search for
+     * @param {String} haystack which might contain needle
+     * @return {Bool}
+     */
+    stringIsPartOfDefineOrRequireStatement(needle, haystack) {
+        const anyCharacterOrNewLine = '.|\\r|\\n';
+        const defineOrRequire = 'define|require';
+        const regex = new RegExp(`(${defineOrRequire})\\((${anyCharacterOrNewLine})*${needle}(${anyCharacterOrNewLine})*(\\)\\.|{)`, 'g');
+        return regex.test(haystack);
+    }
+
+    /**
+     * Get text containing word in range. Text is stripped of comments 
+     * if text contains multiple define/require statements only module part is returned
+     * or if its is part of define/require only the statement 
+     * 
+     * @param {VSCODE Document} document 
+     * @param {VSCODE Range} range 
+     * @return {String} text
+     */
+    getTextContainingWord(document, range) {
+        const text = this.stripAllComments(document.getText());
+        const textAtCaret = document.getText(range);
+
+        let textToParse = text;
+
+        if (this.stringHasMultipleDefineOrRequireStatements(text)) {
```

Comment:
OK, I'll create a new branch for you and rebase these changes onto it. When it is finished it will be merged into master If we won't have a huuge diff. Is it OK for you?